src/VisualStudio/Core/Def/Implementation/Workspace/VisualStudioTaskSchedulerFactory.cs

```diff
@@ -30,6 +30,11 @@ public VisualStudioTaskSchedulerFactory(IThreadingContext threadingContext, IAsy
 
         public override IWorkspaceTaskScheduler CreateEventingTaskQueue()
         {
+            if (!_threadingContext.HasMainThread)
+            {
+                return new DenyWorkTaskScheduler();
```

Comment:
This should just throw, or otherwise we're just moving the crash from the point where we noticed something was wrong (and could debug it) to the point where it's going to cause random failures in other places.

Reply:
This cannot throw. It's part of the initialization sequence and would invalidate operations that never use the queue.

Reply:
What operations don't use the queue?

Reply:
This change is now removed from this pull request.

Reply:
:memo: I looked into just throwing here, but it's not clean. VisualStudioTaskSchedulerFactory may or may not be imported as a unit with VisualStudioWorkspace, so throwing here due to a requirement of the latter would not be clear.